### DISCUSSION TITLE
How to sync auto update database with prisma?

### DISCUSSION THREAD

**Discussion by rahXephonz** [prisma/prisma#19495] [Q&A] [answered]
Hai,

I'm using Prisma for connect to my databases (POSTGRES)
but in some cases when i update for example row username in table User then i re querying the data
function like this

```ts
getAllServices: async (
    _: AnyType,
    {
      limit,
      offset,
      isQueryMyServices,
    }: { isQueryMyServices: boolean } & TPagination,
    { db, req }: Context
  ) => {
    const isGetMyServices =
      isQueryMyServices && req.userId
        ? { isPublished: true, sellerId: req.userId }
        : { isPublished: true };

    const findAllServices = await db.service.findMany({
      where: isGetMyServices,
      orderBy: [{ createdAt: "desc" }],
      include: {
        images: true,
        seller: true,
        ratings: true,
        orders: true,
      },
      take: limit,
      skip: offset,
    });

    const allServicesWithRatings = findAllServices.map((service) => {
      return {
        ...service,
        orderCompleted: filter(service.orders, { isCompleted: true }).length,
        totalRatings: service.ratings.length,
        totalValueRatings: getTotalValueRatings(service.ratings),
      };
    });

    return allServicesWithRatings;
  },
  ```
  
  its still using outdated data from database expect it must be new data from what i update before in db. When debug this cause by a server from express which if any update on code or server the server will restart and the query data will be resync.
  
My question is, how do you query data, whether it's old data or new data? no matter server refresh(restart) or not, keep query up to date?

Any help please?

**Comment by nurul3101** [comment]
Hello @ioofy 👋 

It's difficult to predict what could be the issue for getting stale/outdated data. Are you using any cache?
Can you check that data does actually get updated when you invoke an update query? 
You can check through a database IDE like TablePlus or DataGrip or by inspecting your data through [Prisma Studio](https://www.prisma.io/studio).

**Reply by nurul3101** [reply]
@ioofy Checking to see if you had a chance to look at my last response?

**Reply by rahXephonz** [reply]
Hi thanks for answer my QA, No i mean if i after update let say i update user mutation using GraphQL and then i fetching data the query getAllProducts which have relation with user using

‘’’ts
include : {
   seller: true // this mean seller refers to User schema
}
‘’’
And when see query the user data still using outdated data.

Like that

**Reply by nurul3101** [reply]
Would you be able to provide a minimal reproduction so that I can test it out and check what could be the issue?

**Reply by rahXephonz** [reply]
Hi nurul! sure i can you can see my project in here and running locally in your machine

https://github.com/ioofy/openartist_backend/tree/master/src/services/mutation

Thanks a lot for help!

**Reply by nurul3101** [reply]
Thanks for sharing the link, I can access the repository. The issue is in this query right?
https://github.com/ioofy/openartist_backend/blob/master/src/services/query/serviceQuery.ts#L22

**Reply by rahXephonz** [reply]
yeah, so how about the issue?

**Reply by rahXephonz** [reply, marked as answer]
Oops i think issue in caching when using loader data..
i've solve that with updating my caching method

Sorry..
